src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -137,7 +137,11 @@ BlockIO InterpreterDropQuery::executeToTableImpl(ContextPtr context_, ASTDropQue
                 table_id.getNameForLogs());
 
         /// Now get UUID, so we can wait for table data to be finally dropped
-        table_id.uuid = database->tryGetTableUUID(table_id.table_name);
+        /// If case insensitivity setting is enabled, perform the lookup without case sensitivity
+        if (context_->getSettingsRef().enable_case_insensitive_tables)
+            table_id.uuid = database->tryGetTableUUIDCaseInsensitive(table_id.table_name);
+        else
+            table_id.uuid = database->tryGetTableUUID(table_id.table_name);
```

Comment:
We should not allow case-insensitive names in DDL queries because `DDLGuard` will not work properly:
https://github.com/ClickHouse/ClickHouse/blob/9382c6ffdcc77bd1105fe0cc7565089bd5bb32e4/src/Interpreters/DatabaseCatalog.h#L143-L144